tests/test_telemetry.py

```diff
@@ -324,22 +322,20 @@ def test_app_heartbeat(self):
         # first heartbeat from the list
         heartbeats.pop(0)
 
-        for data in heartbeats:
-            curr_message_time = datetime.strptime(data["request"]["timestamp_start"], fmt)
-            if prev_message_time is None:
-                logger.debug(f"Heartbeat in {data['log_filename']}: {curr_message_time}")
-            else:
-                delta = (curr_message_time - prev_message_time).total_seconds()
-                logger.debug(f"Heartbeat in {data['log_filename']}: {curr_message_time} => {delta}s ellapsed")
+        timestamps = [datetime.strptime(data["request"]["timestamp_start"], fmt) for data in heartbeats]
 
-                assert (
-                    delta < UPPER_LIMIT
-                ), f"Heartbeat sent too slow ({delta}s). It should be sent every {expected_heartbeat_interval}s"
-                assert (
-                    delta > LOWER_LIMIT
-                ), f"Heartbeat sent too fast ({delta}s). It should be sent every {expected_heartbeat_interval}s"
+        # In theory, it's sorted. Let be safe
+        timestamps.sort()
 
-            prev_message_time = curr_message_time
+        delta = (timestamps[-1] - timestamps[0]).total_seconds()
```

Comment:
NIT: it's minor since the test is going to fail either way, but I think you should assert that there are at least two timestamps in the collection.

Reply:
it's already tested few line above. Though there is an issue, as the first timestamp is remove. Let me fix that